src/v/pandaproxy/error.cc

```diff
@@ -229,12 +263,78 @@ std::error_condition make_error_condition(ss::httpd::reply::status_type st) {
     rec ec{};
 
     switch (st) {
-    case sec::forbidden:
-        ec = reply_error_code::kafka_authorization_error;
+    case sec::bad_request:
+        ec = reply_error_code::kafka_bad_request;
         break;
     case sec::unauthorized:
         ec = reply_error_code::kafka_authentication_error;
         break;
+    case sec::forbidden:
+        ec = reply_error_code::forbidden;
+        break;
+    case sec::not_found:
+        ec = reply_error_code::not_found;
+        break;
+    case sec::method_not_allowed:
+        ec = reply_error_code::method_not_allowed;
+        break;
+    case sec::not_acceptable:
+        ec = reply_error_code::not_acceptable;
+        break;
+    case sec::use_proxy:
+        ec = reply_error_code::use_proxy;
+        break;
+    case sec::request_timeout:
+        ec = reply_error_code::request_timeout;
+        break;
+    case sec::conflict:
+        ec = reply_error_code::conflict;
+        break;
+    case sec::gone:
+        ec = reply_error_code::gone;
+        break;
+    case sec::length_required:
+        ec = reply_error_code::length_required;
+        break;
+    case sec::payload_too_large:
+        ec = reply_error_code::payload_too_large;
+        break;
+    case sec::uri_too_long:
+        ec = reply_error_code::uri_too_long;
+        break;
+    case sec::unsupported_media_type:
+        ec = reply_error_code::unsupported_media_type;
+        break;
+    case sec::unprocessable_entity:
+        ec = reply_error_code::unprocessable_entity;
+        break;
+    case sec::upgrade_required:
+        ec = reply_error_code::upgrade_required;
+        break;
+    case sec::too_many_requests:
+        ec = reply_error_code::too_many_requests;
+        break;
+    case sec::internal_server_error:
+        ec = reply_error_code::internal_server_error;
+        break;
+    case sec::not_implemented:
+        ec = reply_error_code::not_implemented;
+        break;
+    case sec::bad_gateway:
+        ec = reply_error_code::bad_gateway;
+        break;
+    case sec::service_unavailable:
+        ec = reply_error_code::service_unavailable;
+        break;
+    case sec::gateway_timeout:
+        ec = reply_error_code::gateway_timeout;
+        break;
+    case sec::http_version_not_supported:
+        ec = reply_error_code::http_version_not_supported;
+        break;
+    case sec::insufficient_storage:
+        ec = reply_error_code::insufficient_storage;
+        break;
     default:
         ec = reply_error_code::kafka_bad_request;
```

Comment:
Usually in switch statements it's useful to not handle the default. That way, under maintenance of the enum, the compiler can tell you if something new was added.

It looks like some might be missing. In particular, it's probably worth mapping `sec::ok` to `reply_error_code{}`, rather than `kafka_bad_request`.

Reply:
> Usually in switch statements it's useful to not handle the default. That way, under maintenance of the enum, the compiler can tell you if something new was added.

That default was there already but I can remove it.

> It looks like some might be missing. In particular, it's probably worth mapping `sec::ok` to `reply_error_code{}`, rather than `kafka_bad_request`.

Do we need to account for non 4xx/5xx error codes here? Those seem unrelated.
Also, I mapped all errors from Seastar https://docs.seastar.io/master/structseastar_1_1httpd_1_1reply.html#aa21b00d3269a13c3b852fcdb0c1c30b3 and those don't have error codes for 412, 416-421, 423-425, 427, or 428.